File: Python/Easy/141-linked-list-cycle/linked-list-cycle.py
class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        if not head:
            return False

        ponteiro_rapido = head 
        ponteiro_lento = head

        while ponteiro_rapido is not None and ponteiro_rapido.next is not None:
            ponteiro_lento = ponteiro_lento.next
            ponteiro_rapido = ponteiro_rapido.next.next

            if ponteiro_rapido == ponteiro_lento:
                return True

        return False

File: 141-linked-list-cycle/linked-list-cycle.cs
public class Solution {
    public bool HasCycle(ListNode head) {
        if(head == null || head.next == null){
            return false;
        }

        ListNode ponteiro_lento = head;
        ListNode ponteiro_rapido = head;

        while(ponteiro_rapido != null && ponteiro_rapido.next != null){
            ponteiro_lento = ponteiro_lento.next;
            ponteiro_rapido = ponteiro_rapido.next.next;

            if(ponteiro_lento == ponteiro_rapido){
                return true;
            }
        }
        return false;
    }
}